test/TritonGPU/amd/amd-canonicalize-pointers.mlir

```diff
@@ -389,6 +389,57 @@ module attributes {"ttg.num-warps" = 4 : i32} {
 
 // -----
 
+module attributes {"ttg.num-warps" = 4 : i32} {
+  tt.func @ifOpTwoYields(%arg0: !tt.ptr<f32>, %arg1: tensor<1024xf32>, %arg2: i1) -> (tensor<1024xf32>, tensor<1024xf32>) {
+    %c1024_i32 = arith.constant 1024 : i32
+    %0 = tt.get_program_id x : i32
+    %1 = arith.muli %0, %c1024_i32 : i32
+    %2 = tt.make_range {end = 1024 : i32, start = 0 : i32} : tensor<1024xi32>
+    %3 = tt.splat %1 : i32 -> tensor<1024xi32>
+    %4 = arith.addi %3, %2 : tensor<1024xi32>
+    %5 = tt.splat %arg0 : !tt.ptr<f32> -> tensor<1024x!tt.ptr<f32>>
+    %6:2 = scf.if %arg2 -> (tensor<1024x!tt.ptr<f32>>, tensor<1024x!tt.ptr<f32>>) {
```

Comment:
can you also test the case where there is a non pointer value as well in the returns? (maybe add a 3rd return value). Looking at the code I'm not sure how it would work

Reply:
I annotate the yields with `kSCFIfOpYieldFatPtrOffsets` which keeps track of exactly where the pointers are and actually yea I need to not lose the value ranges for those values as well (missing those right now).

Reply:
Added `ifOpTwoYieldsAndNonPtr` and `ifOpTwoYieldsAndNonPtrReordered`